### PR TITLE
feat: add personal App credential bootstrap E2E

### PR DESCRIPTION
**Pull request by aydabd**
## Summary

Add a guarded personal-account GitHub App credential bootstrap and E2E path:

- Generate a GitHub App Manifest URL and convert GitHub-generated credentials.
- Authorize a personal App user token through OAuth or Device Flow.
- Deliver only the client ID variable and repository secrets through protected setup commands.
- Exercise personal public/private repository creation with fail-closed owner, token, visibility, cleanup, and organization-token contracts.
- Keep organization installation-token E2E pending because no disposable organization is available.

## Related Issue

No linked issue. Vault/OIDC remains out of scope under issue #92 and is intentionally not closed by this PR.

## Validation

- [x] Deterministic App manifest, user-token, installer, personal E2E, and auth contract tests pass.
- [x] Shell syntax checks pass: `bash -n scripts/github-setup/*.sh`
- [x] Complete lint passes: `LINT_MODE=check make lint`
- [x] GitHub CI passes: quality, CodeQL Actions, and CodeQL Python.
- [x] Generated-file or template parity is verified when applicable (not applicable to these local helpers and workflow additions).
- [ ] Credentialed personal-account workflow E2E is pending until the merged workflow is available on the default branch.
- [ ] Organization installation-token E2E is pending until disposable organization access exists.

## Review Checklist

- [x] Scope is limited to one concern.
- [x] No secrets or mutable action references were added.
- [x] Documentation and acceptance criteria are up to date.
- [x] Commit includes a Signed-off-by trailer.

### CODE REVIEW

File: scripts/github-setup/github-app-user-token.sh [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        while [ "$(date +%s)" -lt "$deadline" ]; do
+            poll_file="$(mktemp)"
+            chmod 600 "$poll_file"
+            trap 'rm -f "$poll_file"' EXIT
+            curl --fail --silent --show-error --location \
+                --header 'Accept: application/json' \
+                --data-urlencode "client_id=$client_id" \
+                --data-urlencode "device_code=$device_code" \
+                --data-urlencode 'grant_type=urn:ietf:params:oauth:grant-type:device_code' \
+                'https://github.com/login/oauth/access_token' > "$poll_file"
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-user-token.sh]: Fixed in 1bf1c83.  now creates one mode-0600 response file before the polling loop and installs one EXIT cleanup trap, so repeated polls cannot accumulate orphaned temporary files.

File: scripts/github-setup/github-app-user-token.sh [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        fi
+        response_file="$(mktemp)"
+        trap 'rm -f "$response_file"' EXIT
+        chmod 600 "$response_file"
+        curl --fail --silent --show-error --location \
+            --header 'Accept: application/json' \
+            --data-urlencode "client_id=$client_id" \
+            --data-urlencode "client_secret=$APP_CLIENT_SECRET" \
+            --data-urlencode "redirect_uri=$redirect_uri" \
+            --data-urlencode "code$code_field" \
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-user-token.sh]: Fixed in 1bf1c83. OAuth exchange now requires  and passes it to curl as ; the secret is not loaded into a shell variable or included in process argv.

File: scripts/github-setup/github-app-manifest.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+}
+
+manifest_json() {
+    local app_name="${1:-github-bootstrap-personal-e2e}"
+    local redirect_url="${2:-https://github.com/settings/apps/new}"
+    cat <<EOF
+{"name":"$app_name","url":"https://github.com","redirect_url":"$redirect_url","public":false,"default_events":[],"default_permissions":{"administration":"write","contents":"write","issues":"write","metadata":"read"}}
+EOF
+}
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-manifest.sh]: Fixed in 1bf1c83. Manifest construction now uses Python  before URL encoding. The contract test exercises quote and newline input and verifies the exact fixed permission set remains intact.

File: scripts/github-setup/github-app-user-token.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-user-token.sh]: Fixed in 1bf1c83. device-poll now creates one mode-0600 response file before the polling loop and installs one EXIT cleanup trap, so repeated polls cannot accumulate orphaned temporary files.

File: scripts/github-setup/github-app-user-token.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-user-token.sh]: Fixed in 1bf1c83. OAuth exchange now requires APP_CLIENT_SECRET_FILE and passes it to curl with data-urlencode client_secret at file. The secret is not loaded into a shell variable or included in process argv.

File: scripts/github-setup/github-app-manifest.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-manifest.sh]: Fixed in 1bf1c83. Manifest construction now uses Python json.dumps before URL encoding. The contract test exercises quote and newline input and verifies the exact fixed permission set remains intact.

File: scripts/github-setup/test-personal-app-e2e-contract.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+repo_root="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
+workflow="$repo_root/.github/workflows/test-personal-app-e2e.yml"
+
+grep -Fq 'BOOTSTRAP_APP_CLIENT_ID' "$workflow"
+grep -Fq 'BOOTSTRAP_APP_PRIVATE_KEY' "$workflow"
+grep -Fq 'BOOTSTRAP_APP_USER_TOKEN' "$workflow"
+grep -Fq 'create-repository.yml' "$workflow"
+grep -Fq 'public' "$workflow"
+grep -Fq 'private' "$workflow"
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/test-personal-app-e2e-contract.sh]: Fixed in 422c15e. The contract now requires the exact workflow entries visibility: public and visibility: private, preventing unrelated text from satisfying the scenario checks.

File: docs/superpowers/plans/2026-08-25-app-manifest-personal-e2e.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
+# GitHub App Manifest Personal E2E Implementation Plan
+
+> **For agentic workers:** REQUIRED SUB-SKILL: Use superpowers:subagent-driven-development (recommended) or superpowers:executing-plans to implement this plan task-by-task. Steps use checkbox (`- [ ]`) syntax for tracking.
```
[the remark this answers is not in the document]
Comment (aydabd) [reply docs/superpowers/plans/2026-08-25-app-manifest-personal-e2e.md]: Fixed in 422c15e. The plan now states that workflow skills come from the active agent environment and no longer claims repository-local required sub-skills.

File: scripts/github-setup/test-personal-app-e2e-contract.sh [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+grep -Fq 'create-repository.yml' "$workflow"
+grep -Fq 'visibility: public' "$workflow"
+grep -Fq 'visibility: private' "$workflow"
+grep -Fq 'gh api --method DELETE' "$workflow"
+grep -Fq 'PERSONAL_OWNER' "$workflow"
+grep -Fq 'allowed_repo_owners' "$workflow"
+if grep -Eq 'gh_token:|GH_PAT|workflow_dispatch.*token|organization-administration' "$workflow"; then
+    echo "personal E2E must not accept PATs, credential inputs, or organization permissions" >&2
+    exit 1
+fi
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/test-personal-app-e2e-contract.sh]: Fixed in 65f8594. The contract now extracts the workflow_dispatch block and rejects credential-shaped input keys at the input indentation level, so multiline YAML inputs cannot bypass the check.

File: README.md [hunk 7]
```diff
@@ -67,6 +67,42 @@ the App user access token, verifies its `/user` login against the target owner,
...
+0700 output directory, with each file set to 0600. The example keeps that directory outside the
+repository checkout; remove it after setup. Use the App client ID and secret to authorize the
+personal account, then exchange the callback code:
+
+```bash
+credential_dir="$HOME/.local/state/github-bootstrap"
+scripts/github-setup/github-app-user-token.sh url CLIENT_ID OWNER REDIRECT_URI STATE
+APP_CLIENT_SECRET_FILE="$credential_dir/app-client-secret" \
+APP_REDIRECT_URI=https://github.com/settings/apps/new \
+  scripts/github-setup/github-app-user-token.sh exchange CLIENT_ID CODE OWNER "$credential_dir/app-user-token"
```
[the remark this answers is not in the document]
Comment (aydabd) [reply README.md]: Fixed in 8950ef6. The README now defines one redirect_uri value and reuses it for both the authorization URL and APP_REDIRECT_URI during token exchange.

File: .github/workflows/test-personal-app-e2e.yml [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+          PUBLIC_REPO: bootstrap-app-e2e-${{ github.run_id }}-public
+          PRIVATE_REPO: bootstrap-app-e2e-${{ github.run_id }}-private
+        shell: bash
+        run: |
+          set -euo pipefail
+          if ! [[ "$PERSONAL_OWNER" =~ ^[A-Za-z0-9]([A-Za-z0-9-]{0,37}[A-Za-z0-9])?$ ]]; then
+            echo "Invalid personal owner; refusing cleanup API calls" >&2
+            exit 1
+          fi
+          for repository in "$PUBLIC_REPO" "$PRIVATE_REPO"; do
```
[the remark this answers is not in the document]
Comment (aydabd) [reply .github/workflows/test-personal-app-e2e.yml]: Fixed in ffa7d54. Cleanup now validates owner syntax, confirms /users/$PERSONAL_OWNER has type User, and verifies /user identity matches PERSONAL_OWNER before checking or deleting either repository.

File: scripts/github-setup/github-app-user-token.sh [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+            }
+            [ -n "${APP_CLIENT_SECRET_FILE:-}" ] && [ -f "$APP_CLIENT_SECRET_FILE" ] || {
+                echo "APP_CLIENT_SECRET_FILE must name a protected App client secret file" >&2
+                exit 1
+            }
+            [ -n "$client_id" ] && [ -n "$code_or_file" ] && [ -n "$owner" ] && [ -n "$token_file" ] && [ -n "$redirect_uri" ] || usage
+            response_file="$(mktemp)"
+            chmod 600 "$response_file"
+            sanitized_code_file=""
+            trap 'rm -f "$response_file" "$sanitized_code_file"' EXIT
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-user-token.sh]: Fixed in 351b20d. The EXIT trap now removes the response file unconditionally and removes the sanitized code file only when it was created, avoiding rm -f on an empty path.

File: scripts/github-setup/install-app-secrets.sh [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    echo "client ID file is empty" >&2
+    exit 1
+}
+case "$token" in
+    ghu_*) ;;
+    *)
+        echo "user token file must contain a GitHub App user token with ghu_ prefix" >&2
+        exit 1
+        ;;
+esac
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/install-app-secrets.sh]: Fixed in 399463d. The installer now calls the token’s /user endpoint with GH_TOKEN and compares the normalized login to the owner extracted from OWNER/REPO before writing any repository variable or secret.

File: README.md [hunk 11]
```diff
@@ -67,6 +67,43 @@ the App user access token, verifies its `/user` login against the target owner,
...
+credential_dir="$HOME/.local/state/github-bootstrap"
+scripts/github-setup/github-app-manifest.sh url
+# Open the printed URL, approve the App, and copy the one-time conversion code.
+scripts/github-setup/github-app-manifest.sh convert CODE "$credential_dir"
+```
+
+The conversion command writes GitHub's private key, client ID, and client secret only under the
+0700 output directory, with each file set to 0600. The example keeps that directory outside the
+repository checkout; remove it after setup. Use the App client ID and secret to authorize the
+personal account, then exchange the callback code:
```
[the remark this answers is not in the document]
Comment (aydabd) [reply README.md]: Fixed in 399463d. README now documents deleting BOOTSTRAP_APP_PRIVATE_KEY, BOOTSTRAP_APP_USER_TOKEN, and BOOTSTRAP_APP_CLIENT_ID after the disposable E2E, plus revoking or rotating the App credentials and removing the local directory.

File: scripts/github-setup/install-app-secrets.sh [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+# shellcheck source=scripts/github-setup/gh-common.sh
+source "$script_dir/gh-common.sh"
+
+usage() {
+    cat >&2 << 'EOF'
+Usage: install-app-secrets.sh REPOSITORY CLIENT_ID_FILE PRIVATE_KEY_FILE USER_TOKEN_FILE
+
+Installs the GitHub-generated private key and ghu_-prefixed App user token as
+repository secrets, and the client ID as a repository variable. Token owner
+identity must be verified before invoking this installer.
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/install-app-secrets.sh]: Fixed in 9e87f68. The installer help now states that it verifies token owner identity before writing repository values.

File: scripts/github-setup/github-app-user-token.sh [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+                    esac
+                    authenticated_user="$(GH_TOKEN="$token" gh api /user --jq '.login')"
+                    normalized_user="$(printf '%s' "$authenticated_user" | tr '[:upper:]' '[:lower:]')"
+                    normalized_owner="$(printf '%s' "$owner" | tr '[:upper:]' '[:lower:]')"
+                    if [ "$normalized_user" != "$normalized_owner" ]; then
+                        echo "App user token identity does not match the requested personal owner." >&2
+                        exit 1
+                    fi
+                    printf '%s' "$token" > "$token_file"
+                    chmod 600 "$token_file"
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-user-token.sh]: Fixed in 9e87f68. Token output now writes to a mode-0600 temporary file in the destination directory and atomically renames it into place, avoiding direct writes through existing paths or symlinks.

File: scripts/github-setup/github-app-user-token.sh [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+The exchange command verifies the token's /user identity and writes only the
+short-lived access token to TOKEN_FILE with mode 0600. Other OAuth response
+credentials are never written or printed.
+EOF
+    exit 2
+}
+
+main() {
+    command_name="${1:-}"
+    case "$command_name" in
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-user-token.sh]: Fixed in 06a2f69. The user-token helper now dispatches through a small main function and separate device-start, device-poll, URL, and exchange functions.

File: scripts/github-setup/github-app-user-token.sh [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+                if [[ ! "$code_or_file" =~ ^[A-Za-z0-9_-]+$ ]]; then
+                    echo "invalid OAuth authorization code" >&2
+                    exit 1
+                fi
+                code_field="=$code_or_file"
+            fi
+            curl --fail --silent --show-error --location \
+                --header 'Accept: application/json' \
+                --data-urlencode "client_id=$client_id" \
+                --data-urlencode "client_secret@$APP_CLIENT_SECRET_FILE" \
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-user-token.sh]: Fixed in 06a2f69. The client secret is normalized into a mode-0600 temporary file with CR/LF removed, and curl reads that file without placing the secret in argv. The temporary file is cleaned up on exit.

File: scripts/github-setup/install-app-secrets.sh [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+fi
+
+sanitized_token_file="$(mktemp)"
+chmod 600 "$sanitized_token_file"
+trap 'rm -f "$sanitized_token_file"' EXIT
+printf '%s' "$token" > "$sanitized_token_file"
+
+gh variable set BOOTSTRAP_APP_CLIENT_ID --repo "$repo" --body "$client_id"
+gh secret set BOOTSTRAP_APP_PRIVATE_KEY --repo "$repo" < "$private_key_file"
+gh secret set BOOTSTRAP_APP_USER_TOKEN --repo "$repo" < "$sanitized_token_file"
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/install-app-secrets.sh]: Fixed in b97f0e5. The installer strips CR characters into a mode-0600 sanitized private-key file and uploads that normalized file to the repository secret.

File: scripts/github-setup/github-app-user-token.sh [hunk 17]
```diff
@@ -0,0 +1,237 @@
...
+    local token_value="$1"
+    local output_path="$2"
+    local output_dir
+    local output_base
+    local temporary_path
+
+    output_dir="$(dirname "$output_path")"
+    output_base="$(basename "$output_path")"
+    temporary_path="$(mktemp "$output_dir/.$output_base.XXXXXX")"
+    chmod 600 "$temporary_path"
```
[the remark this answers is not in the document]
Comment (aydabd) [reply scripts/github-setup/github-app-user-token.sh]: Fixed in b97f0e5. write_protected_token now checks that the destination directory exists and reports a clear error before calling mktemp.

File: docs/superpowers/specs/2026-08-25-app-manifest-personal-e2e-design.md [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+
+## Design
+
+Add a local shell helper that creates a GitHub App Manifest authorization URL and a second command that exchanges the one-time conversion code with GitHub. The conversion response is stored only under a caller-selected 0700 directory, with the returned private key, client ID, and client secret written to separate 0600 files. This allows the caller to put the GitHub-generated private key into `BOOTSTRAP_APP_PRIVATE_KEY` without logging it.
+
+Add a separate authorization helper that accepts the App client ID as an argument and the client secret through a protected file path, starts the GitHub App user OAuth flow, exchanges the callback code, verifies `/user` is the intended personal owner, and emits token metadata without the token value. Secret delivery remains an explicit `gh secret set` operation from protected files.
+
+Add contract tests for manifest permissions, secret boundaries, token prefix/identity checks, and the absence of workflow-dispatch credential inputs. Add a credentialed personal E2E entry point that requires an explicitly enabled environment and never falls back to PAT authentication.
```
[the remark this answers is not in the document]
Comment (aydabd) [reply docs/superpowers/specs/2026-08-25-app-manifest-personal-e2e-design.md]: Fixed in b97f0e5. The design spec now describes the actual manual-dispatch guard and explicit personal-owner input instead of claiming a GitHub Actions environment requirement.
